library/src/main/java/fr/tvbarthel/intentshare/TargetActivity.java

```diff
@@ -82,6 +137,39 @@ public Uri getIconUri() {
      * @return true if the target activity is a mail client, false otherwise.
      */
     public boolean isMailClient() {
-        return isMail;
+        return this.isMail;
+    }
+
+    /**
+     * Comparator used to sort {@link TargetActivity} based on the recency of their previous
+     * selection and their name as fallback when they have never been selected.
+     */
+    public static class RecencyComparator implements Comparator<TargetActivity> {
```

Comment:
This ```RecencyComparator``` is inconsistent with ```TagetActivity#equals()```, it might be a good idea to mention it in the comments =)

https://docs.oracle.com/javase/7/docs/api/java/util/Comparator.html